packages/account-v2/src/utils/poiUtils.ts

```diff
@@ -18,3 +26,70 @@ export const setErrorMessage = <T extends TSocketError<TSocketEndpointNames>['er
             return message ?? API_ERROR_CODES.generic.message;
     }
 };
+
+export const translateErrorCode = (errorCode: string | null, service: TPOIService) => {
+    if (!errorCode) {
+        return '';
+    }
+    if (service === POI_SERVICE.idv) {
+        return (
+            Object.values(IDV_ERROR_CODES).find(error => error.code === errorCode)?.message ??
+            IDV_ERROR_CODES.generic.message
+        );
+    }
+    return (
+        Object.values(ONFIDO_ERROR_CODES).find(error => error.code === errorCode)?.message ??
+        ONFIDO_ERROR_CODES.generic.message
+    );
+};
+
+type TIDVErrorStatusConfig = {
+    errors?: string[];
+    isReportAvailable?: 0 | 1;
+    status?: TPOIStatus;
+};
+
+export const checkIDVErrorStatus = ({
+    errors,
+    isReportAvailable,
+    status,
+}: TIDVErrorStatusConfig): TIDVErrorStatusCode | null => {
+    if (status === AUTH_STATUS_CODES.EXPIRED) {
+        return IDV_ERROR_CODES.expired.code;
+    }
+
+    if (!errors?.length) {
+        return null;
+    }
+
+    if (isReportAvailable === 0) {
+        return IDV_ERROR_CODES.reportNotAvailable.code;
+    }
+    if (
+        errors.includes(IDV_ERROR_CODES.nameMismatch.code as string) &&
+        errors.includes(IDV_ERROR_CODES.dobMismatch.code as string)
```

Comment:
```suggestion
        errors.some(error => error === IDV_ERROR_CODES.nameMismatch.code || error === IDV_ERROR_CODES.nameMismatch.code)
```

will it help to avoid type assertion? @likhith-deriv